introspection build fails when default library is static

Guess that is intended? @ebassi @fanc999 not asking for a fix but just to know is there anything release blocker here? Thanks

```
$ meson sbuild --default-library=static -Dintrospection=enabled && ninja -Csbuild
ninja: Entering directory 'sbuild'
[154/285] Generating HarfBuzz-0.0.gir with a custom command.
FAILED: src/HarfBuzz-0.0.gir 
/usr/bin/g-ir-scanner -I/usr/include/gobject-introspection-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -pthread --no-libtool --namespace=HarfBuzz --nsversion=0.0 --warn-all --output src/HarfBuzz-0.0.gir --c-include=hb-gobject.h --cflags-begin -DHB_H -DHB_H_IN -DHB_OT_H -DHB_OT_H_IN -DHB_AAT_H -DHB_AAT_H_IN -DHB_GOBJECT_H -DHB_GOBJECT_H_IN -DHB_EXTERN= --cflags-end -I/path/harfbuzz/src -I/path/harfbuzz/sbuild/src -I./. -I../. -I./. -I../. --filelist=/path/harfbuzz/sbuild/src/25a6634@@harfbuzz@sta/HarfBuzz_0.0_gir_filelist --include=GObject-2.0 --symbol-prefix=hb --symbol-prefix=hb_gobject --identifier-prefix=hb_ --pkg-export=harfbuzz-gobject --cflags-begin -fno-rtti -fno-exceptions -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -DHAVE_CONFIG_H -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include --cflags-end --library /path/harfbuzz/sbuild/src/libharfbuzz.a -L/path/harfbuzz/sbuild/src --library /path/harfbuzz/sbuild/src/libharfbuzz-gobject.a -L/path/harfbuzz/sbuild/src -L/path/harfbuzz/sbuild/src --extra-library=m --extra-library=freetype --extra-library=glib-2.0 --extra-library=gobject-2.0 --sources-top-dirs /path/harfbuzz/subprojects/ --sources-top-dirs /path/harfbuzz/sbuild/subprojects/
g-ir-scanner: link: clang -o /path/harfbuzz/sbuild/tmp-introspectc0ir9yfi/HarfBuzz-0.0 /path/harfbuzz/sbuild/tmp-introspectc0ir9yfi/HarfBuzz-0.0.o -L. -Wl,-rpath,. -Wl,--no-as-needed -L/path/harfbuzz/sbuild/src -Wl,-rpath,/path/harfbuzz/sbuild/src -L/path/harfbuzz/sbuild/src -Wl,-rpath,/path/harfbuzz/sbuild/src -L/path/harfbuzz/sbuild/src -Wl,-rpath,/path/harfbuzz/sbuild/src /path/harfbuzz/sbuild/src/libharfbuzz.a /path/harfbuzz/sbuild/src/libharfbuzz-gobject.a -lm -lfreetype -lglib-2.0 -lgobject-2.0 -lgio-2.0 -lgobject-2.0 -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -lglib-2.0
/usr/bin/ld: /path/harfbuzz/sbuild/src/libharfbuzz-gobject.a(hb-gobject-structs.cc.o): undefined reference to symbol 'hb_shape_plan_reference'
/usr/bin/ld: /usr/lib/libharfbuzz.so.0: error adding symbols: DSO missing from command line
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
/path/harfbuzz/src/hb-aat-layout.h:36: syntax error, unexpected identifier in 'HB_BEGIN_DECLS' at 'HB_BEGIN_DECLS'
/path/harfbuzz/src/hb-aat-layout.h:431: syntax error, unexpected '*', expecting ')' or ',' in 'hb_aat_layout_get_feature_types (hb_face_t *face,' at '*'
/path/harfbuzz/src/hb-aat-layout.h:433: syntax error, unexpected UNSIGNED, expecting identifier or '(' or '*' in '     unsigned int *feature_count, /* IN/OUT.  May be NULL. */' at 'unsigned'
/path/harfbuzz/src/hb-aat-layout.h:434: syntax error, unexpected typedef-name, expecting identifier or '(' or '*' in '     hb_aat_layout_feature_type_t *features /* OUT.     May be NULL. */);' at 'hb_aat_layout_feature_type_t'
/path/harfbuzz/src/hb-aat-layout.h:434: syntax error, unexpected ')', expecting ',' or ';' in '     hb_aat_layout_feature_type_t *features /* OUT.     May be NULL. */);' at ')'
/path/harfbuzz/src/hb-aat-layout.h:442: syntax error, unexpected identifier in '  hb_ot_name_id_t name_id;' at 'hb_ot_name_id_t'
/path/harfbuzz/src/hb-aat-layout.h:447: syntax error, unexpected '}' in '} hb_aat_layout_feature_selector_info_t;' at '}'
/path/harfbuzz/src/hb-aat-layout.h:452: syntax error, unexpected '*', expecting ')' or ',' in 'hb_aat_layout_feature_type_get_selector_infos (hb_face_t *face,' at '*'
/path/harfbuzz/src/hb-aat-layout.h:454: syntax error, unexpected UNSIGNED, expecting identifier or '(' or '*' in '            unsigned int start_offset,' at 'unsigned'
/path/harfbuzz/src/hb-aat-layout.h:455: syntax error, unexpected UNSIGNED, expecting identifier or '(' or '*' in '            unsigned int *selector_count, /* IN/OUT.  May be NULL. */' at 'unsigned'
/path/harfbuzz/src/hb-aat-layout.h:456: syntax error, unexpected '*', expecting ',' or ';' in '            hb_aat_layout_feature_selector_info_t *selectors, /* OUT.     May be NULL. */' at '*'
/path/harfbuzz/src/hb-aat-layout.h:457: syntax error, unexpected ')', expecting ',' or ';' in '            unsigned int *default_index /* OUT.     May be NULL. */);' at ')'
/path/harfbuzz/src/hb-ot-shape.h:47: syntax error, unexpected '*', expecting ')' or ',' in 'hb_ot_shape_plan_collect_lookups (hb_shape_plan_t *shape_plan,' at '*'
/path/harfbuzz/src/hb-ot-shape.h:49: syntax error, unexpected typedef-name, expecting identifier or '(' or '*' in '      hb_set_t *lookup_indexes /* OUT */);' at 'hb_set_t'
/path/harfbuzz/src/hb-ot-shape.h:49: syntax error, unexpected ')', expecting ',' or ';' in '      hb_set_t *lookup_indexes /* OUT */);' at ')'
../src/hb-buffer.cc:677: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-face.cc:184: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-font.cc:522: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-font.cc:1369: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-map.cc:43: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-ot-name.cc:55: Warning: HarfBuzz: unexpected annotation: out
../src/hb-set.cc:43: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-shape-plan.cc:166: Warning: HarfBuzz: unknown annotation: xconstructor
../src/hb-unicode.cc:157: Warning: HarfBuzz: unknown annotation: xconstructor
linking of temporary binary failed: Command '['clang', '-o', '/path/harfbuzz/sbuild/tmp-introspectc0ir9yfi/HarfBuzz-0.0', '/path/harfbuzz/sbuild/tmp-introspectc0ir9yfi/HarfBuzz-0.0.o', '-L.', '-Wl,-rpath,.', '-Wl,--no-as-needed', '-L/path/harfbuzz/sbuild/src', '-Wl,-rpath,/path/harfbuzz/sbuild/src', '-L/path/harfbuzz/sbuild/src', '-Wl,-rpath,/path/harfbuzz/sbuild/src', '-L/path/harfbuzz/sbuild/src', '-Wl,-rpath,/path/harfbuzz/sbuild/src', '/path/harfbuzz/sbuild/src/libharfbuzz.a', '/path/harfbuzz/sbuild/src/libharfbuzz-gobject.a', '-lm', '-lfreetype', '-lglib-2.0', '-lgobject-2.0', '-lgio-2.0', '-lgobject-2.0', '-Wl,--export-dynamic', '-lgmodule-2.0', '-pthread', '-lglib-2.0', '-lglib-2.0']' returned non-zero exit status 1.
[163/285] Compiling C++ object 'src/25...@@harfbuzz-subset@sta/hb-subset.cc.o'.
ninja: build stopped: subcommand failed.
```

--- a/src/meson.build
+++ b/src/meson.build
@@ -617,7 +617,17 @@ if have_gobject
   )
 
   gir = find_program('g-ir-scanner', required: get_option('introspection'))
-  build_gir = gir.found() and not meson.is_cross_build()
+  build_gir = gir.found()
+
+  build_gir = build_gir and not meson.is_cross_build()
+  if not build_gir and get_option('introspection').enabled()
+    error('Introspection support is requested but it isn\'t available in cross builds')
+  endif
+
+  build_gir = build_gir and get_option('default_library') != 'static'
+  if not build_gir and get_option('introspection').enabled()
+    error('Introspection support is requested but the default library option should be shared or both')
+  endif
 
   if build_gir
     conf.set('HAVE_INTROSPECTION', 1)

